### PR TITLE
Added support for Zmtel greenpacket devices (os: zmtel)

### PR DESCRIPTION
**Pull request by spencerbutler**
Fixes:
#10065
#8729

I'm not entirely sure about the Frequencies. The device reports in KHz and LibreNMS uses Hz. I multiplied by 1000, but it doesn't seem correct. Please advise.

>ulFrequency OBJECT-TYPE
  SYNTAX  OCTET STRING (SIZE (0..16))
        MAX-ACCESS read-only
        STATUS  current
        DESCRIPTION "Indicates the mobile network uplink frequency(KHz)."
        ::= { lteInfo 9 }

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by inkweb** [comment]
So i did "./scripts/github-apply 10037" as the librenms user, but the devices are still showing up as "generic device"

**Comment by PipoCanaja** [comment]
@inkweb 
```
rm cache/os_defs.cache 
````
That will ensure the new patch is used.

**Comment by inkweb** [comment]
That worked, thanks.

So for the frequencies, here's what LibreNMS is reporting:

> Bandwidth 		20000000Hz
> Download 		3612500000Hz
> Upload 		3612500000Hz

And this is what the CPE itself says:

> DL Frequency 3612500 KHz
> UL Frequency 3612500 KHz
> Bandwidth 20000 KHz

Everything else looks good so far. There are some items missing that we require to monitor the units:

- SINR
- CINR0
- CINR1
- PCI
- Dl MCS
- Ul MCS

If those could be added as well, that would be wonderful.

**Comment by spencerbutler** [comment]
> That worked, thanks.
> 
> So for the frequencies, here's what LibreNMS is reporting:
> 
> > Bandwidth 		20000000Hz
> > Download 		3612500000Hz
> > Upload 		3612500000Hz
> 
> And this is what the CPE itself says:
> 
> > DL Frequency 3612500 KHz
> > UL Frequency 3612500 KHz
> > Bandwidth 20000 KHz
> 
> Everything else looks good so far. There are some items missing that we require to monitor the units:
> 
> * SINR
> * CINR0
> * CINR1
> * PCI
> * Dl MCS
> * Ul MCS
> 
> If those could be added as well, that would be wonderful.

Great. I'll fix up the frequencies and see if I can add the requested data points.

**Comment by spencerbutler** [comment]
@inkweb git this latest commit a try and let me know what you think.

**Comment by murrant** [comment]
@inkweb those are the same frequencies...

**Comment by spencerbutler** [comment]
> This PR is abusing the count sensor.
> 
> Please do not do that.

fixed

**Comment by inkweb** [comment]
> @inkweb those are the same frequencies...

yes i know, i'm just letting everyone know what i'm seeing on this end as there was some question about the frequency handling in the update.

**Comment by inkweb** [comment]
Frequencies, SINR, CINR0 and CINR1 are all looking great, thank you! Any luck with PCI and the two MCS datas?

**Comment by spencerbutler** [comment]
> Any luck with PCI and the two MCS datas?

Those were moved to the "Operating System" line in the "Overview" section.

**Comment by inkweb** [comment]
Ah i see. Any way to make them graphable? It would help with troubleshooting.

**Comment by spencerbutler** [comment]
Do those values change over time? If so, I'll look into the proper way to graph them.

**Comment by inkweb** [comment]
Yes they do change

File: includes/polling/os/zmtel.inc.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+$temp_data = snmpwalk_cache_oid($device, 'deviceInfo', [], 'ZMTEL-ODU-MIB', 'zmtel');
```
Comment (laf) [inline includes/polling/os/zmtel.inc.php]: Can you not just do a multi get here for the 4 values you need rather than working additional data?

Comment (spencerbutler) [reply includes/polling/os/zmtel.inc.php]: fixed

**Comment by inkweb** [comment]
MCS (and everything else so far) seems to be working great. Graphing PCI would be great too, so we can find people tower jumping.

**Comment by spencerbutler** [comment]
> MCS (and everything else so far) seems to be working great. Graphing PCI would be great too, so we can find people tower jumping.

It should be graphing now.

**Comment by inkweb** [comment]
It is graphing, but the graph is kind of [odd](https://i.imgur.com/cn7EMTV.jpg), with the line flat at the bottom and the whole scale on the left being a single number.

**Comment by inkweb** [comment]
but its not a deal breaker by any means. basically we're fine with the device as it stands now. thank you for the work!

**Comment by spencerbutler** [comment]
> but its not a deal breaker by any means. basically we're fine with the device as it stands now. thank you for the work!

Awesome! Glad to hear it's working for you.

**Comment by inkweb** [comment]
Sorry, one more thing, when doing an API request of the devices, is it possible to get it to return PCI as a key?

**Pull request merged**